lib/OptimizationEvolutionary/src/OptimizationEvolutionary.jl

```diff
@@ -11,9 +11,30 @@ SciMLBase.supports_opt_cache_interface(opt::Evolutionary.AbstractOptimizer) = tr
 decompose_trace(trace::Evolutionary.OptimizationTrace) = last(trace)
 decompose_trace(trace::Evolutionary.OptimizationTraceRecord) = trace
 
-function Evolutionary.trace!(record::Dict{String, Any}, objfun, state, population,
-        method::Evolutionary.AbstractOptimizer, options)
```

Comment:
This method would not be dispatched if user defined own method more specific than `AbstractOptimizer`, allowing `x` to not be saved and throwing an error quite opaque to the user, as none of this is exposed